addons/delivery/models/stock_move.py

```diff
@@ -2,12 +2,27 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from odoo import api, fields, models
-
+from odoo.tools.sql import column_exists, create_column
 
 
 class StockMove(models.Model):
     _inherit = 'stock.move'
 
+    def _auto_init(self):
+        if not column_exists(self.env.cr, "stock_move", "weight"):
+            # In case of a big database with a lot of stock moves, the RAM gets exhausted
+            # To prevent a process from being killed We create the column 'weight' manually
+            # Then we do the computation in a query by multiplying product weight with qty
+            create_column(self.env.cr, "stock_move", "weight", "numeric")
+            self.env.cr.execute("""
+                UPDATE stock_move move
+                SET weight = move.product_uom_qty * product.weight
```

Comment:
You should use product_qty instead of product_uom_qty to have the same UoM than the product